src/linux.cc

```diff
@@ -318,30 +318,30 @@ int update_gateway_info2(void) {
   unsigned int flags;
   unsigned int x = 1;
   unsigned int z = 1;
-  int strcmpreturn;
+  unsigned int skip = 0;
 
   if((fp = check_procroute()) != nullptr) {
     while (!feof(fp)) {
-      strcmpreturn = 1;
       if (fscanf(fp, RT_ENTRY_FORMAT, iface, &dest, &gate, &flags, &mask) != 5) {
         update_gateway_info_failure("fscanf()");
         break;
       }
-      if (!(dest || mask) && ((flags & RTF_GATEWAY) || !gate)) {
```

Comment:
It's a little unclear to me from the code why these lines were removed. Could you explain?

Reply:
Didn't have much time this week, but the weekend is here. I'm now starting to check again how I can get the checks to pass without breaking code

Reply:
@brndnmtthws Because the same code is used in 

https://github.com/brndnmtthws/conky/blob/master/src/linux.cc#L330-L332

And it's called here

https://github.com/brndnmtthws/conky/blob/master/src/linux.cc#L442-L444

There is no point overwriting the same address to `e_iface` in both functions.

Reply:
Nothing at all works here correct anymore :-(
I'll wait until it works fine again and then I'll just create a garo-master branch where I can make sure it passes the 3 tests without breaking anything